django_cron/cron.py

```diff
@@ -7,48 +7,94 @@
+            self.check_for_failures(cron_cls)
+
+    def get_config(self):
+        """
+        Combine the default configuration with any project-specific ones.
+        """
+        defaults = dict(
+            FAILED_RUNS_CRONJOB_EMAIL_PREFIX='[Cron Failure] - ',
+            CRON_MIN_NUM_FAILURES=10,
+            CRON_FAILURE_FROM_EMAIL=settings.DEFAULT_FROM_EMAIL,
+            CRON_FAILURE_EMAIL_RECIPIENTS=[
+                email for _, email in settings.ADMINS
+            ]
+        )
+        return {
+            key: getattr(settings, key, defaults[key])
+            for key in defaults
+        }
+
+    def check_for_failures(self, cron_cls):
+        """
+        Check the given Cron task for failed jobs, and report if required.
+        """
+        min_failures = getattr(
+            cron_cls, 'MIN_NUM_FAILURES', self.config['CRON_MIN_NUM_FAILURES']
+        )
 
-        CRONS_TO_CHECK = map(lambda x: get_class(x), settings.CRON_CLASSES)
-        EMAILS = [admin[1] for admin in settings.ADMINS]
+        failed_jobs = CronJobLog.objects.filter(
+            code=cron_cls.code, is_success=False, failure_reported=False
+        )
 
-        try:
-            FAILED_RUNS_CRONJOB_EMAIL_PREFIX = settings.FAILED_RUNS_CRONJOB_EMAIL_PREFIX
-        except:
-            FAILED_RUNS_CRONJOB_EMAIL_PREFIX = ''
+        if failed_jobs.count() < min_failures:
+            return
 
-        for cron in CRONS_TO_CHECK:
+        self.report_failure(cron_cls, failed_jobs)
+        failed_jobs.update(failure_reported=True)
 
-            try:
-                min_failures = cron.MIN_NUM_FAILURES
-            except AttributeError:
-                min_failures = 10
+    def report_failure(self, cron_cls, failed_jobs):
+        """
+        Report the failed jobs by sending an email (using django-common).
+        """
+        send_mail(**self.get_send_mail_kwargs(cron_cls, failed_jobs))
 
-            failures = 0
+    def get_send_mail_kwargs(self, cron_cls, failed_jobs):
+        """
+        Return the arguments to pass to send_mail for the given failed jobs.
+        """
+        failed_reports = []
 
-            jobs = CronJobLog.objects.filter(code=cron.code).order_by('-end_time')[:min_failures]
+        for job in failed_jobs:
+            failed_reports.append(
+                u"Job ran at {start_time}:\n{message}"
+                .format(start_time=job.start_time, message=job.message)
+            )
 
-            message = ''
+        divider = "\n\n{0}\n\n".format("=" * 80)
+        message = divider.join(failed_reports)
+        subject = "{prefix}{code} failed".format(
+            prefix=self.config['FAILED_RUNS_CRONJOB_EMAIL_PREFIX'],
+            code=cron_cls.code
+        )
 
-            for job in jobs:
-                if not job.is_success:
-                    failures += 1
-                    message += 'Job ran at %s : \n\n %s \n\n' % (job.start_time, job.message)
+        if len(failed_reports) > 1:
```

Comment:
shouldn't this always be true / == to the min_failures? Why not just define times before first definition of subject and add to it there?